tests/Basics/VelocyPackHelperCompareRandomTest.cpp

```diff
@@ -0,0 +1,381 @@
+  } else if constexpr (type == NumberValueType::UInt) {
+    reusableBuilder.add(Value(significand, static_cast<ValueType>(type)));
+  } else if constexpr (type == NumberValueType::Int) {
+    if (signum == -1 && significand == 0) [[unlikely]] {
+      // there is no -0 in two's complement; use ::lowest() instead, which
+      // is also not yet part of the generated values, as its representation
+      // is 1 bit longer.
+      reusableBuilder.add(Value(std::numeric_limits<std::int64_t>::lowest(),
+                                static_cast<ValueType>(type)));
+    } else {
+      reusableBuilder.add(Value(signum * static_cast<std::int64_t>(significand),
+                                static_cast<ValueType>(type)));
+    }
+  } else if constexpr (type == NumberValueType::SmallInt) {
+    reusableBuilder.add(Value(signum * static_cast<std::int32_t>(significand),
+                              static_cast<ValueType>(type)));
+  }
+  std::memcpy(vpBuffer, reusableBuilder.slice().start(),
+              reusableBuilder.slice().byteSize());
+}
+
+thread_local Builder VPackHelperRandomTest::reusableBuilder;
+
+TEST_F(VPackHelperRandomTest, test_me) {
+  auto left = Number{
+      .significand = 34359738368,
+      .exponent = 0,
+      .signum = -1,
+      .type = NumberValueType::Int,
+  };  // slice=-34359738368
+  left.writeVPack<NumberValueType::Int>();
+  auto right = Number{
+      .significand = 264904856511168,
+      .exponent = -13,
+      .signum = -1,
+      .type = NumberValueType::Double,
+  };  // slice=-32337018617.085938
+  right.writeVPack<NumberValueType::Double>();
+  EXPECT_EQ(left <=> right, std::weak_ordering::less);
+  EXPECT_EQ(right <=> left, std::weak_ordering::greater);
+}
+
+// TODO maybe add a handful of tests for Number::operator<=>
+
+static auto compareVPack(VPackHelperRandomTest::Number const& left,
+                         VPackHelperRandomTest::Number const& right)
+    -> std::strong_ordering {
+  auto res = basics::VelocyPackHelper::compareNumberValuesCorrectly(
+      left.slice().type(), left.slice(), right.slice());
+  if (res < 0) {
+    return std::strong_ordering::less;
+  }
+  if (res > 0) {
+    return std::strong_ordering::greater;
+  }
+  return std::strong_ordering::equal;
+}
+
+static auto compareAqlValue(VPackHelperRandomTest::Number const& left,
+                            VPackHelperRandomTest::Number const& right)
+    -> std::strong_ordering {
+  velocypack::Options options;
+  arangodb::aql::AqlValue leftValue{left.slice()};
+  arangodb::aql::AqlValue rightValue{right.slice()};
+  auto res =
+      arangodb::aql::AqlValue::Compare(&options, leftValue, rightValue, true);
+  if (res < 0) {
+    return std::strong_ordering::less;
+  }
+  if (res > 0) {
+    return std::strong_ordering::greater;
+  }
+  return std::strong_ordering::equal;
+}
+
+TEST_F(VPackHelperRandomTest, test_vpackcmps) {
+  SCOPED_TRACE(fmt::format("seed={}", seed));
+  constexpr auto num = 10000;
```

Comment:
this will perform 10000 * 10000 * 2 = 200M comparisons.
what's the expected runtime on relevant hardware? Is this rather a few seconds or is this going to run much slower? In this case, we may want to assign each random number to one of `x` buckets, and then run the checks concurrently for each bucket using multiple threads.

Reply:
its like 15s on my laptop 🤔 I don't think that this is too much